api/v1alpha1/wavefront_webhook_test.go

```diff
@@ -0,0 +1,55 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"testing"
+)
+
+func TestDefault(t *testing.T) {
+	spec := WavefrontSpec{
+		WavefrontUrl:          "testWavefrontUrl",
+		WavefrontToken:        "testToken",
+		ClusterName:           "",
+		CollectorEnabled:      true,
+		WavefrontProxyEnabled: true,
+		ProxyUrl:              "",
+		ControllerManagerUID:  "",
+	}
+	var r = Wavefront{
+		TypeMeta:   metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{},
+		Spec:       spec,
+		Status:     WavefrontStatus{},
+	}
+	expectedClusterName := "k8s-cluster"
+	r.Default()
+	if r.Spec.ClusterName == "" {
+		t.Errorf("Expected spec ClusterName to not be empty.")
+	}
+	if r.Spec.ClusterName != expectedClusterName {
+		t.Errorf("Expected default clusterName :: %s, but got %s", expectedClusterName, r.Spec.ClusterName)
+	}
+}
+
+func TestValidateWavefront(t *testing.T) {
+	spec := WavefrontSpec{
+		WavefrontUrl:          "",
+		WavefrontToken:        "",
+		ClusterName:           "",
+		CollectorEnabled:      true,
+		WavefrontProxyEnabled: true,
+		ProxyUrl:              "",
+		ControllerManagerUID:  "",
+	}
+	var r = Wavefront{
+		TypeMeta:   metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{},
+		Spec:       spec,
+		Status:     WavefrontStatus{},
+	}
+	err := r.validateWavefront()
```

Comment:
I get why you wanted to test `validateWavefront` directly instead of writing one test for create and another for update. However, I prefer to test public methods only. The fact that create and update happen to share the same validation seems like an implementation detail that we shouldn't couple our tests to. I'd consider the two tests using public methods.